### PR TITLE
Converted some things to the correct types

### PR DESCRIPTION
**Pull request by PMunch**
To work with the fixed version of Gtk2 in pull request [nim-lang/gtk2#22](https://github.com/nim-lang/gtk2/pull/22).

### CODE REVIEW

File: search.nim [hunk 1]
```diff
@@ -305,7 +305,7 @@ proc findText*(win: var utils.MainWin, forward: bool) =
     highlightAll(win, pattern, true, win.autoSettings.search)
   else:
     # Stop it from highlighting due to selection.
-    win.tabs[currentTab].highlighted = newHighlightAll("", true, -1)
+    win.tabs[currentTab].highlighted = newHighlightAll("", true, guint.high)
```
Comment (dom96) [inline search.nim]: Why is this change required?

File: utils.nim [hunk 2]
```diff
@@ -188,7 +188,7 @@ type
     compileSuccess*: bool
     execThread*: Thread[void]
     execProcess*: Process
-    idleFuncId*: int32
+    idleFuncId*: guint#int32
```
Comment (dom96) [inline utils.nim]: Remove the comment.

File: search.nim (hunk 1 above)
Comment (dom96) [reply search.nim]: If something down the line needs a guint you should change it there, not up here.

File: utils.nim (hunk 2 above)
Comment (PMunch) [reply utils.nim]: Done

File: search.nim (hunk 1 above)
Comment (PMunch) [reply search.nim]: Well, the newHighlightAll procedure simply sets the field idleID to the applied value. idleID is used in one procedure call, and obtained from another, both being glib procedures which expects a guint. So there's really no reason why this should be represented as an int in the Nim code just to be converted every time it's actually used. The reason why guint.high is required is simply that Nim is a bit more restrictive than C which is fine with assigning -1 to a uint. In Nim you can't do this without explicitly casting, but the value of -1 cast to a guint and guint.high should be the same.

Comment (dom96) [reply search.nim]: Meh, alright.
